L1Trigger/L1TCaloLayer1/plugins/L1TCaloSummary.cc

```diff
@@ -172,7 +168,7 @@ void L1TCaloSummary::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
   // independently creating regions from TPGs for processing by the summary card. This results
   // in a single region vector of size 252 whereas from independent creation we had 3*6 vectors
   // of size 7*2. Indices are mapped in UCTSummaryCard accordingly.
-  summaryCard->clearRegions();
+  summaryCard = new UCTSummaryCard(&pumLUT, jetSeed, tauSeed, tauIsolationFactor, eGammaSeed, eGammaIsolationFactor);
```

Comment:
too bad the first review didn't suggest to use smart pointers here...

but at this point, why not just make this a local variable?
```
UCTSummaryCard summaryCard(&pumLUT, jetSeed, tauSeed...);
```

Reply:
I seem to recall looking at this problem a while ago, and trying to make changes to smart pointers, running into some thing that explicitly expected a pointer, and quitting before I ended up needing to make changes to a bunch of crucial L1 Emulator legacy code. 

However, looking again, I can't find where that would be an issue in this code, and I don't see a reason why it couldn't just be a local variable either. 

I'll make the change.

Reply:
I am very probably suggesting this in the wrong place, but in the future, I think an implementation here: 

https://github.com/cms-sw/cmssw/blob/b2a6f4eb6ba99575db3f89df0e6fac09e4e6ba8b/L1Trigger/L1TCaloLayer1/src/UCTSummaryCard.cc#L502-L505

that explicitly frees up this memory (to my knowledge a `std::vector` `.clear()` does not explicitly `delete` the contents) would circumvent the need for this particular change